src/gui/qgsdoublevalidator.h

```diff
@@ -0,0 +1,144 @@
+#ifndef QGSDOUBLEVALIDATOR_H
+#define QGSDOUBLEVALIDATOR_H
+
+/***************************************************************************
+                         qgsdoublevalidator.h  -  description
+                             -------------------
+    begin                : June 2020
+    copyright            : (C) 2020 by Sebastien Peillet
+    email                : sebastien.peillet@oslandia.com
+
+  adapted version of Qgslonglongvalidator + QgsFieldValidator
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <limits>
+#include <QRegExpValidator>
+#include <QLocale>
+#include "qgis_gui.h"
+
+/**
+ * \ingroup gui
+ * \class QgsDoubleValidator
+ * \since 3.14
+ */
+class GUI_EXPORT QgsDoubleValidator : public QRegExpValidator
+{
+    Q_OBJECT
+
+  public:
+    explicit QgsDoubleValidator( QObject *parent )
+      : QRegExpValidator( parent )
+      , b( std::numeric_limits<qreal>::min() )
+      , t( std::numeric_limits<qreal>::max() )
+    {
+      // The regular expression accept double with point as decimal point but also the locale decimal point
+      QRegExp reg( QStringLiteral( "-?[\\d]{0,1000}([\\.%1][\\d]{0,1000})?([\\d]e[+-]?[\\d]{0,1000})?" ).arg( locale().decimalPoint() ) );
+      this->setRegExp( reg );
+    }
+
+    QgsDoubleValidator( QRegExp reg, double bottom, double top, QObject *parent )
+      : QRegExpValidator( parent )
+      , b( bottom )
+      , t( top )
+    {
+      this->setRegExp( reg );
+    }
+
+    QgsDoubleValidator( double bottom, double top, QObject *parent )
+      : QRegExpValidator( parent )
+      , b( bottom )
+      , t( top )
+    {
+      // The regular expression accept double with point as decimal point but also the locale decimal point
+      QRegExp reg( QStringLiteral( "-?[\\d]{0,1000}([\\.%1][\\d]{0,1000})?([\\d]e[+-]?[\\d]{0,1000})?" ).arg( locale().decimalPoint() ) );
```

Comment:
If the expression is the same you should turn it into a constant (uppercase) and place it at the top of the file.

Reply:
:+1: Done, I placed it in the new cpp file